src/vs/platform/files/browser/indexedDBFileSystemProvider.ts

```diff
@@ -427,7 +428,18 @@ export class IndexedDBFileSystemProvider extends Disposable implements IFileSyst
 	private async writeMany() {
 		if (this.fileWriteBatch.length) {
 			const fileBatch = this.fileWriteBatch.splice(0, this.fileWriteBatch.length);
-			await this.indexedDB.runInTransaction(this.store, 'readwrite', objectStore => fileBatch.map(entry => objectStore.put(entry.content, entry.resource.path)));
+			try {
+				await this.indexedDB.runInTransaction(this.store, 'readwrite', objectStore => fileBatch.map(entry => {
+					return objectStore.put(entry.content, entry.resource.path);
+				}));
+			} catch (ex) {
+				const estimated = await navigator?.storage?.estimate?.();
```

Comment:
Came across this exception name [QuotaExcededError](https://webidl.spec.whatwg.org/#quotaexceedederror). Is not enough to check if the extension is `QuotaExcededError`?